CHANGELOG.md

```diff
@@ -1,8 +1,487 @@
 # go-ipfs changelog
 
-## 0.4.20-rc2 2019-04-11
+## 0.4.20 2019-04-16
 
-TODO
+We're happy to release go-ipfs 0.4.20. This release includes some critical
+performance and stability fixes so all users should upgrade ASAP.
+
+This is also the first release to use go modules instead of GX. While GX has
+been a great way to dogfood an IPFS-based package manager, building and
+maintaining a custom package manager is a _lot_ of work and we haven't been able
+to dedicate enough time to bring the user experience of gx to an acceptable
+level. You can read [#5850](https://github.com/ipfs/go-ipfs/issues/5850) for
+some discussion on this matter.
+
+### Docker
+
+As of this release, it's now much easier to run arbitrary IPFS commands within
+the docker container:
+
+```bash
+> docker run ipfs/go-ipfs config profile apply server # apply the server profile
+> docker run ipfs/go-ipfs # start the daemon
+```
+
+This release also [reverts](https://github.com/ipfs/go-ipfs/pull/6040) a change that
+caused some significant trouble in 0.4.19. If you've been running into Docker
+permission errors in 0.4.19, please upgrade.
+
+### WebUI
+
+This release contains a major
+[WebUI](https://github.com/ipfs-shipyard/ipfs-webui) release with some
+significant improvements to the file browser and new opt-in, privately hosted,
+anonymous usage analytics.
+
+### Commands
+
+As usual, we've made several changes and improvements to our commands. The most
+notable changes are listed in this section.
+
+#### New: `ipfs version deps`
+
+This release includes a new command, `ipfs version deps`, to list all
+dependencies (with versions) of the current go-ipfs build. This should make it
+easy to tell exactly how go-ipfs was built when tracking down issues.
+
+#### New: `ipfs add URL`
+
+The `ipfs add` command has gained support for URLs. This means you can:
+
+1. Add files with `ipfs add URL` instead of downloading the file first.
+2. Replace all uses of the `ipfs urlstore` command with a call to `ipfs add
+   --nocopy`. The `ipfs urlstore` command will be deprecated in a future
+   release.
+
+
+#### Changed: `ipfs swarm connect`
+
+The `ipfs swarm connect` command has a few new features:
+
+It now marks the newly created connection as "important". This should ensure
```

Comment:
cc @jacobheun @vasco-santos

Reply:
I'll be looking at adding tagging support in js this quarter for connection management, so this is something we should be able to look at adding in q3.

Reply:
As I noted above, this _shouldn't_ really be necessary. However, we're still killing useful connections so this change makes the issue slightly less annoying.